include/fmt/format.h

```diff
@@ -70,9 +70,9 @@
 #endif
 
 #if __cplusplus == 201103L || __cplusplus == 201402L
-#  if defined(__clang__)
+#  if defined(__clang__) && !defined(__INTEL_COMPILER)
```

Comment:
I suggest moving the Intel compiler check in a branch of its own, something like

```c++
#  ifdef __INTEL_COMPILER
#    define FMT_FALLTHROUGH
#  elif ...
```
to avoid repetition.

Reply:
how about something like this? looks like PGI falls into the same category.

Reply:
LGTM